.github/actions/check-dependant-rust-repo/action.yml

```
@@ -0,0 +1,125 @@
+      git checkout uprev-test
+      git log -1 --format='%H'
+      cd ..
+
+      # Try to build
+      ${{ inputs.buildCmd }} && {
+        # Build succeeded, lets see if this is a new development.
+        if [ -f "$STATE_FILE" ]; then
+          echo "We previously failed building, but this run succeeded :)"
+          rm $STATE_FILE
+
+          # This lets the steps below know that we fixed a previously failed build.
+          echo "UPREV_FIXED=1" >> "$GITHUB_ENV"
+        fi
+       } || {
+        # We failed building, so  keep track of that in the env. This lets the steps below
+        # take action based on whether we failed or not.
+        echo "UPREV_FAILED=1" >> "$GITHUB_ENV"
+
+        # If this a new failure?
+        if [ -f "$STATE_FILE" ]; then
+          echo "Not the first time this branch failed, so we won't be posting an update."
+          echo "NEW_FAILURE=0" >> "$GITHUB_ENV"
+        else
+          echo "Detected a new failure, indicating by writing to $STATE_FILE"
+          echo > $STATE_FILE
+          echo "NEW_FAILURE=1" >> "$GITHUB_ENV"
+        fi
+      }
+
+  # A step that tries to build the remote repository with the upstream branch of the current PR.
+  # We do this since if it fails, we can't blame the failure on the changes in the current PR
+  # and there is no point in posting about the failure (but we will post if this PR fixes the build).
+  - name: Build with base branch (${{ github.base_ref }})
+    if: env.UPREV_FAILED == '1' && env.NEW_FAILURE == '1'
+    shell: bash
+    run: |
+      cd workdir-${{ inputs.id }}
+
+      cd ${{ inputs.submodulePath }}
+      #git remote set-url origin ${{ github.repositoryUrl }} - this is git:// and GHA doesn't seem to like it
+      git remote set-url origin https://github.com/${{ github.repository }}
+      git fetch origin ${{ github.base_ref }}:gha-test-base
+      git checkout gha-test-base
+      cd ..
+
+      ${{ inputs.buildCmd }} && echo "BASE_FAILED=0" >> "$GITHUB_ENV" || echo "BASE_FAILED=1" >> "$GITHUB_ENV"
+
+  # TODO it would be nice if this message included a link to the log with the failed build
+  - name: Comment on PR (failure)
+    uses: mshick/add-pr-comment@v1
+    if: env.UPREV_FAILED == '1' && env.BASE_FAILED == '0' && env.NEW_FAILURE == '1'
+    with:
+      message: |
+        Heads up, I tried building ${{ inputs.id }} using this branch and it failed.
+      repo-token: ${{ inputs.githubToken }}
+      repo-token-user-login: 'github-actions[bot]' # The user.login for temporary GitHub tokens
+      allow-repeats: true # This is the default
+
+  - name: Comment on PR (success)
+    uses: mshick/add-pr-comment@v1
+    if: env.UPREV_FIXED == '1'
+    with:
+      message: |
+        ✅ Good job, ${{ inputs.id }} was built successfully.
+      repo-token: ${{ inputs.githubToken }}
+      repo-token-user-login: 'github-actions[bot]' # The user.login for temporary GitHub tokens
+      allow-repeats: true # This is the default
```

Consider adding a step that fails this action if UPREV_FAILED, so we get a check failure
```suggestion
      allow-repeats: true # This is the default

  - name: Propagate build status
    shell: bash
    run: exit ${UPREV_FAILED:-0}
```

---

I intentionally didn't make it fail this action since I don't think the PR status should be red if this fails - more often than not the changes that break the outside repo are perfectly fine, we just need to go update the outside repo at some point to accommodate for them.
Perhaps tagging the person that opened the PR in the comment that this makes in the PR might make that comment more visible? (if this is possible)